providers/zone.rb

```diff
@@ -23,11 +23,34 @@ def whyrun_supported?
 
 use_inline_resources
 
+def load_current_resource
+  @current_resource = Chef::Resource::T3PdnsZone.new(new_resource.name)
+
+  @current_resource.name(new_resource.name)
+  @current_resource.exists = ::File.file?("/etc/powerdns/zones/#{new_resource.name}.zone")
+  @current_resource
+end
+
+# Returns the pdns_control reload command, depending on if
+# we create a new zone or update an existing one.
+def reload_command
+
+  if @current_resource.exists
+    log "Zone #{@new_resource.name} already exists. Doing reload."
+    "pdns_control bind-reload-now #{@new_resource.name}"
+  else
+    log "New zone #{@new_resource.name}. Adding."
+    "pdns_control bind-add-zone #{@new_resource.name} /etc/powerdns/zones/#{new_resource.name}.zone"
+  end
```

Comment:
`pdns_control reload` does not recognize new zones, only updated ones. In my tests, this worked. @andristeiner do you expect any bad side-effects by changing this?